Implemented the temperature sensor on the ADC chip for analogRead(-1) on gen1

anthony-ngu opened the pull request:
The readTemp could be combined with the readValue using a couple if, else statements.
One for the channel being 8 for Tsense
One for the sleep in order to provide enough time to elapse for the conversion.

I chose to make them separate initially for debugging purposes, but then chose to leave it for clarity. This can easily be combined.

Unfortunately the gen2 ADC chip does not seem to provide a temperature sensor that is accessible to us.

ooeygui commented:
Could you fix your tabbing - it looks out of sync with the file.

ooeygui commented on source/AD7298Support.h:
```diff
@@ -53,6 +53,118 @@ class AD7298Device
+
+		// TsenseBusy should be high until the conversion is completed (100 micro seconds max)
+		// wait 100ns after TsenseBusy is low to start reading
+		Sleep(1);
```
Sleep(1) is 1 millisecond. Can we use a different delay?

ooeygui commented on source/AD7298Support.h:
```diff
@@ -53,6 +53,118 @@ class AD7298Device
+
+        // TsenseBusy should be high until the conversion is completed (100 micro seconds max)
+        // wait 100ns after TsenseBusy is low to start reading
+        Sleep(1);
```
Use delay(100) here.

anthony-ngu commented:
Fixed the tabbing. I used a Sleep for 1 ms since delayMicroseconds was not available

anthony-ngu commented:
Fixed the tabbing
ReadTemp is now its own function (separate from analogRead)
Changed the Sleep to WindowsTime.delayMicroseconds
Added documentation about function use
